controllers/whoToFollow/whoToFollowV2.js

```diff
@@ -0,0 +1,267 @@
+          });
+        }
+
+        if (tempUsers.length > 0) {
+          result.push({
+            viewtype: 'labellocation',
+            ...location,
+            users_count: tempUsers.length,
+            users: tempUsers
+          });
+        }
+      }
+    }
+    //END LOCATIONS
+
+    //OTHER USERS
+    if (page == 1 && topics) {
+      const otherTopicsQuery = `SELECT 
+                            a.user_id,
+                            a.country_code,
+                            a.username,
+                            a.real_name,
+                            a.created_at,
+                            a.updated_at,
+                            a.last_active_at,
+                            a.status,
+                            a.profile_pic_path,
+                            a.profile_pic_asset_id,
+                            a.profile_pic_public_id,
+                            a.bio,
+                            a.is_banned,
+                            a.is_anonymous,
+                            a.encrypted,
+                            a.allow_anon_dm,
+                            a.only_received_dm_from_user_following,
+                            a.is_backdoor_user,
+                            a.followers_count,
+                            a.karma_score,
+                            b.topic_id
+                          FROM users a
+                          INNER JOIN user_topics b ON a.user_id = b.user_id
+                          WHERE 
+                            a.is_anonymous = false
+                            AND a.is_banned = false
+                            AND a.blocked_by_admin = false
+                            AND a.karma_score > 30
+                            AND a.user_id != :admin_user_id
+                            AND b.topic_id NOT IN (:topics)
+                          ORDER BY 
+                            a.last_active_at DESC,
+                            CASE WHEN a.profile_pic_path != '%default-profile-picture%' THEN 0 ELSE 1 END,
+                            a.followers_count DESC
+                          LIMIT :limit`;
+
+      const userOtherTopicFollowerQueryResult = await sequelize.query(otherTopicsQuery, {
+        type: sequelize.QueryTypes.SELECT,
+        replacements: {
+          topics,
+          admin_user_id: process.env.BETTER_ADMIN_ID,
+          limit
+        }
+      });
+      const userOtherTopicFollower = userOtherTopicFollowerQueryResult;
+
+      const tempUsers = [];
+      for (const indexUserOtherTopic in userOtherTopicFollower) {
+        const userTopic = userOtherTopicFollower[indexUserOtherTopic];
+
+        tempUsers.push({
+          ...userTopic,
+          viewtype: 'user'
+        });
+      }
+
+      if (tempUsers.length > 0) {
+        result.push({
+          viewtype: 'labelothers',
+          name: 'Other popular members',
+          users_count: tempUsers.length,
+          users: tempUsers
+        });
+      }
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to be repeating the same logic as the blocks for topics and locations. Consider abstracting this logic into a separate function to avoid code duplication and improve maintainability.



<!-- This is an auto-generated comment by OSS CodeRabbit -->